integration/querier_test.go

```diff
@@ -471,38 +470,43 @@ func TestQuerierWithBlocksStorageRunningInSingleBinaryMode(t *testing.T) {
-			// - 1 cache request for each series
+			// - 2 cache requests for each series: one while sending label values and another while streaming chunks. The second request should be a cache hit.
 			//
 			// If the series does not exist in the block, then we return early after checking expanded postings and
 			// subsequently the index on disk.
 			result, err := c.Query(series1Name, series1Timestamp)
 			require.NoError(t, err)
 			require.Equal(t, model.ValVector, result.Type())
 			assert.Equal(t, expectedVector1, result.(model.Vector))
-			expectedCacheRequests += seriesReplicationFactor * 3 // expanded postings, postings, series
+			expectedCacheRequests += seriesReplicationFactor * 4        // expanded postings, postings, series (retrieving labels and then while streaming chunks)
+			expectedCacheHits += seriesReplicationFactor                // one for each series (for chunks)
+			expectedMemcachedOps += seriesReplicationFactor * (3*2 + 1) // Same reasoning as for expectedCacheRequests, but this also includes a set for each get that is not a hit
 
 			result, err = c.Query(series2Name, series2Timestamp)
 			require.NoError(t, err)
 			require.Equal(t, model.ValVector, result.Type())
 			assert.Equal(t, expectedVector2, result.(model.Vector))
-			expectedCacheRequests += seriesReplicationFactor*3 + seriesReplicationFactor // expanded postings, postings, series for 1 time range; only expaned postings for another
+			expectedCacheRequests += seriesReplicationFactor*4 + seriesReplicationFactor // expanded postings, postings, series for 1 time range; only expanded postings for another
+			expectedCacheHits += seriesReplicationFactor                                 // one for each series
+			expectedMemcachedOps += seriesReplicationFactor * (4 + 3 + 1)                // Same reasoning as for expectedCacheRequests, but this also includes a set for each get that is not a hit
 
 			result, err = c.Query(series3Name, series3Timestamp)
 			require.NoError(t, err)
 			require.Equal(t, model.ValVector, result.Type())
 			assert.Equal(t, expectedVector3, result.(model.Vector))
 			expectedCacheRequests += seriesReplicationFactor * 2 // expanded postings for 2 time ranges
-
-			expectedMemcachedOps := 2 * expectedCacheRequests // Same reasoning as for expectedCacheRequests, but this also includes a set for each get
+			expectedMemcachedOps += seriesReplicationFactor * 2
 
 			// Check the in-memory index cache metrics (in the store-gateway).
 			require.NoError(t, cluster.WaitSumMetrics(e2e.Equals(float64(expectedCacheRequests)), "thanos_store_index_cache_requests_total"))
-			require.NoError(t, cluster.WaitSumMetrics(e2e.Equals(0), "thanos_store_index_cache_hits_total")) // no cache hit cause the cache was empty
+			require.NoError(t, cluster.WaitSumMetrics(e2e.Equals(float64(expectedCacheHits)), "thanos_store_index_cache_hits_total")) // no cache hit cause the cache was empty
```

Comment:
```suggestion
			require.NoError(t, cluster.WaitSumMetrics(e2e.Equals(float64(expectedCacheHits)), "thanos_store_index_cache_hits_total"))
```

Reply:
Addressed in [23602e3](https://github.com/grafana/mimir/pull/6646/commits/23602e3b3fc459bce9b793db82569ef2a525ec47)